plyara/command_line.py

```diff
@@ -23,12 +23,15 @@
 from plyara.core import Plyara
 
 
-def main():
+def main(arguments=None):
     """Run the command line process to parse a yara rule file and output pretty printed JSON."""
     parser = argparse.ArgumentParser(description='Parse YARA rules into a dictionary representation.')
     parser.add_argument('file', metavar='FILE', help='File containing YARA rules to parse.')
     parser.add_argument('--log', help='Enable debug logging to the console.', action='store_true')
-    args = parser.parse_args()
+    if not arguments:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(arguments)
```

Comment:
i haven't seen this pattern before, but I like it.

Reply:
This makes it easy to pass fake arguments in the unittest. Using subprocess is frowned upon for security reasons and the new unit test avoids using subprocess, but still tests stdout for the command line output and checks for errors on stderr.